pkg/registry/github_test.go

```diff
@@ -508,3 +511,71 @@ func Test_parseGitHubURI(t *testing.T) {
 		}
 	}
 }
+
+func TestGitHub_CacheRoot(t *testing.T) {
+	defaultURL := "github.com/ksonnet/parts/tree/master/incubator"
+	tests := []struct {
+		name      string
+		url       string
+		path      string
+		expected  string
+		expectErr bool
+	}{
+		{
+			name:     "starts with registry name",
+			url:      defaultURL,
+			path:     "incubator/apache/parts.yaml",
+			expected: "incubator/apache/parts.yaml",
+		},
+		{
+			name:     "doesn't start with registry name",
+			url:      defaultURL,
+			path:     "notincubator/apache/parts.yaml",
+			expected: "incubator/notincubator/apache/parts.yaml",
+		},
+		{
+			name:     "deals with leading slash (unexpected)",
+			url:      defaultURL,
+			path:     "/incubator/apache/parts.yaml",
+			expected: "incubator/apache/parts.yaml",
+		},
+		{
+			name:     "root file",
+			url:      defaultURL,
+			path:     "registry.yaml",
+			expected: "incubator/registry.yaml",
+		},
+		{
+			name:     "registry name and url tail are different",
+			url:      "github.com/ksonnet/parts/tree/master/foobar",
+			path:     "foobar/apache/parts.yaml",
+			expected: "incubator/apache/parts.yaml",
+		},
+		// {
```

Comment:
Will these work at a later date?

Reply:
Only if I bypass the constructor. Is that worth the bother?